2025/incoming/the-gift-of-readability.pod

```diff
@@ -0,0 +1,450 @@
+
+=item * B<include> — Filter-in operation
+
+=back
+
+Avoid verbs like:
+
+=over
+
+=item * B<get> — Too generic. Consult a thesaurus; this verb has dozens of meanings
+
+=item * B<filter> — Inconclusive. Better to use C<include> or C<exclude>
+
+=item * B<maybe> — Such logic is part of the method's contract; the method should decide what to do
+
+=item * B<return> — Redundant when used as a verb in method names
+
+=item * B<sanction> - autoantonym (e.g.: C<sanction_trade_with_russia>)
+
+=back
+
+When everyone speaks the same language, code becomes self-documenting.
+
+For more examples, see: L<https://gist.github.com/happy-barney/1ee36e693751dbbfc046a268cde258de>
+
+It is always a good idea to summarize such verbs and their project meaning in a glossary.
+
+=head2 The Gift of Readability Manifesto
+
+As dawn broke on Christmas morning, she compiled her learnings into a gift for her
+team - The Gift of Readability Manifesto.
+
+=head3 1. Use snake_case Everywhere
+
+Even in package names. C<UserAccountManager> becomes C<User::Account::Manager> or C<User_Account_Manager>,
+variables become C<$user_account_balance>, constants C<USER_ACCOUNT_MANAGER>.
+
+Exercise: play with terms "settable" and "set table".
+
+=head3 2. Indent with Tabs
+
+=over
+
+=item * One tab equals one indentation level. No such things as "half-level".
+
+=item * Do not specify tab width in your project.
+
+=item * Let each contributor configure their editor to their preferred width (2 spaces, 4 spaces, 8 spaces - whatever they like).
+
+Every modern editor supports this.
+
+=back
+
+=head3 3. Never Align on Lines with Different Indentation
+
+=head3 4. Use Paired Characters as Quoted Operator Delimiters
+
+It reduces the number of backslashes you need to use:
+
+=begin code perl
+
+q (function () shouldn't return "$foo");
+qr (a/b);
+
+# vs
+
+'function () shouldn\'t return "$foo"';
+"function () shouldn't return \"\$foo\"";
+qr/a\/b/;
+
+=end code
+
+=head3 5. Space Around Operators and Keywords
+
+Make breathing room:
+
+=begin code perl
+
+if (! $condition) {
+    function_call ($arg =~ m (...));
+}
+
+# Not:
+if(!$condition){
+    function_call($arg=~m/.../);
+}
+
+=end code
+
+Examples of proper and consistent spacing:
+
+=over
+
+=item * C<if ()>, C<function ()>, C<q ()>, C<s () ()>
+
+=item * C<(! condition)>
+
+=item * C<$x = $y + $z>
+
+=back
+
+=head3 6. One Line, One Concept
+
+When writing multiline expressions, each line should contain either:
+
+=over
+
+=item * Complete information (opening and closing parenthesis on the same line), or
+
+=item * A single piece of information (opening parenthesis as the last character)
+
+=back
+
+The left parenthesis is either on the same line as its counterpart or
+is the last character on a given line.
+
+=begin code perl
+
+# Good: complete expression on one line
+my $result = calculate ($param) + other_function ($other);
+
+# Good: clear multiline structure
+my $good
+    = calculate ($param)
```

Comment:
this doesn't quite follow your style though. The `+` is more likely to be notices if both its operands are around it.